types/src/validator_info.rs

```diff
@@ -110,28 +124,49 @@ impl TryFrom<crate::proto::types::ValidatorInfo> for ValidatorInfo {
         let account_address = AccountAddress::try_from(proto.account_address)?;
         let consensus_public_key = Ed25519PublicKey::try_from(&proto.consensus_public_key[..])?;
         let consensus_voting_power = proto.consensus_voting_power;
-        let network_signing_public_key =
-            Ed25519PublicKey::try_from(&proto.network_signing_public_key[..])?;
-        let network_identity_public_key =
-            x25519::PublicKey::try_from(&proto.network_identity_public_key[..])?;
-        Ok(Self::new(
-            account_address,
-            consensus_voting_power,
+
+        let validator_network_signing_public_key =
+            Ed25519PublicKey::try_from(&proto.validator_network_signing_public_key[..])?;
+        let validator_network_identity_public_key =
+            x25519::PublicKey::try_from(&proto.validator_network_identity_public_key[..])?;
+        let validator_network_address = RawNetworkAddress::new(proto.validator_network_address);
+        let full_node_network_identity_public_key =
+            x25519::PublicKey::try_from(&proto.full_node_network_identity_public_key[..])?;
+        let full_node_network_address = RawNetworkAddress::new(proto.full_node_network_address);
+
+        let config = ValidatorConfig::new(
             consensus_public_key,
-            network_signing_public_key,
-            network_identity_public_key,
-        ))
+            validator_network_signing_public_key,
+            validator_network_identity_public_key,
+            validator_network_address,
+            full_node_network_identity_public_key,
+            full_node_network_address,
+        );
+        Ok(Self::new(account_address, consensus_voting_power, config))
     }
 }
 
 impl From<ValidatorInfo> for crate::proto::types::ValidatorInfo {
     fn from(keys: ValidatorInfo) -> Self {
         Self {
             account_address: keys.account_address.to_vec(),
-            consensus_voting_power: keys.consensus_voting_power,
-            consensus_public_key: keys.consensus_public_key.to_bytes().to_vec(),
-            network_signing_public_key: keys.network_signing_public_key.to_bytes().to_vec(),
-            network_identity_public_key: keys.network_identity_public_key.to_bytes(),
+            consensus_voting_power: keys.consensus_voting_power(),
+            consensus_public_key: keys.consensus_public_key().to_bytes().to_vec(),
+            validator_network_signing_public_key: keys
+                .config()
```

Comment:
if we were to make a temporary value on line 129 such that
  let config = &keys.config();

would that let use reduce the bloat here and below?